src/FunctionalArrayCollection.js

```diff
@@ -0,0 +1,69 @@
+import FunctionalCollection from './FunctionalCollection'
+
+/**
+ * Functional collection that works with arrays under the hood
+ */
+export class FunctionalArrayCollection extends FunctionalCollection {
+    /**
+     * Retrieves items of the collection.
+     *
+     * @return {Array} The items, by item ID.
+     */
+    getItems () {
+        return this._getItems().slice()
+    }
+
+    /**
+     * Check that given item in the collection.
+     *
+     * @param item
+     * @returns {boolean}
+     */
+    hasItem (item) {
+        let foundItemIndex = null
+        let items = this.getItems()
+        for (let i in items) {
+            if (this._keyGetter(items[i]) === this._keyGetter(item)) {
+                foundItemIndex = i
+                break
+            }
+        }
+        return foundItemIndex !== null
+    }
+
+    /**
+     * Adds an item to a set.
+     *
+     * @param {Array} items The item set.
+     * @param {Object} item The item to add.
+     *
+     * @return {Object} The modified items set.
+     */
+    _addItem(items, item) {
+        items = this.getItems()
```

Comment:
Also makes a copy of the items set every time you add one.

Reply:
Hmm, I also see it in your implementation [here](https://github.com/RebelCode/std-lib/blob/develop/src/FunctionalCollection.js#L102)

Reply:
Fair enough. Returning new state instead of modifying directly. My bad, this is how it's supposed to be done.

On the other hand, I am making a copy of `items` passed, and you - of some internal items ;)